From 1c906f0d01bf87b8d2c01309f24d35e1e9acc295 Mon Sep 17 00:00:00 2001
From: Rapaka Naveen <naveen.rapaka@intel.com>
Date: Wed, 29 Apr 2015 23:43:11 +0530
Subject: [PATCH] [Revert Me] EM/PMIC: Added delayed workqueue to avoid race
 condition in S3

The pmic ccsm delayed work queue is added.
This was done to avoid the race condtion with i2c resume.

Change-Id: I375c437186cf10f35d5cbe391e1d3c2d1f27bb09
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1504
Signed-off-by: Kumar P, Mahesh <mahesh.kumar.p@intel.com>
Signed-off-by: Rapaka Naveen <naveen.rapaka@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 6 +++---
 drivers/power/intel_pmic_ccsm.h | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 78c09e9..a16dad5 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1169,7 +1169,7 @@ static irqreturn_t pmic_thread_handler(int id, void *data)
 			evt->battemp_int_stat, evt->misc_int,
 			evt->misc_int_stat);
 
-	schedule_work(&chc.evt_work);
+	schedule_delayed_work(&chc.evt_work, msecs_to_jiffies(100));
 	return IRQ_HANDLED;
 }
 
@@ -1511,7 +1511,7 @@ static int pmic_check_initial_events(void)
 
 	INIT_LIST_HEAD(&evt->node);
 	list_add_tail(&evt->node, &chc.evt_queue);
-	schedule_work(&chc.evt_work);
+	schedule_delayed_work(&chc.evt_work, 0);
 
 	pmic_bat_zone_changed();
 
@@ -1747,7 +1747,7 @@ static int pmic_chrgr_probe(struct platform_device *pdev)
 		goto otg_req_failed;
 	}
 
-	INIT_WORK(&chc.evt_work, pmic_event_worker);
+	INIT_DELAYED_WORK(&chc.evt_work, pmic_event_worker);
 	INIT_LIST_HEAD(&chc.evt_queue);
 
 	ret = pmic_check_initial_events();
diff --git a/drivers/power/intel_pmic_ccsm.h b/drivers/power/intel_pmic_ccsm.h
index fca3149..4f9becf 100644
--- a/drivers/power/intel_pmic_ccsm.h
+++ b/drivers/power/intel_pmic_ccsm.h
@@ -398,5 +398,5 @@ struct pmic_chrgr_drv_context {
 	struct usb_phy *otg;
 	struct thermal_cooling_device *vbus_cdev;
 	struct list_head evt_queue;
-	struct work_struct evt_work;
+	struct delayed_work evt_work;
 };
-- 
1.9.1

